### ISSUE TITLE
Error occur while training on VeRi dataset

### ISSUE THREAD

**Issue by irisqp** [nwojke/cosine_metric_learning#31]
@nwojke Thanks for your excellent work! I trained Deep-SORT model on Jeston TX2 , and this error occurs
can you help me figure out why this happens? Thank you very much!
##############################################
Train set size: 34267 images, 519 identites
('feature dimensionality: ', 128)
WARNING:tensorflow:From /home/nvidia/cosine_metric_learning/nets/deep_sort/network_definition.py:84: calling l2_normalize (from tensorflow.python.ops.nn_impl) with dim is deprecated and will be removed in a future version.
Instructions for updating:
dim is deprecated, use axis instead
WARNING:tensorflow:From /home/nvidia/cosine_metric_learning/train_app.py:615: sparse_softmax_cross_entropy (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.sparse_softmax_cross_entropy instead. Note that the order of the logits and labels arguments has been changed.
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/contrib/losses/python/losses/loss_ops.py:434: compute_weighted_loss (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.compute_weighted_loss instead.
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/contrib/losses/python/losses/loss_ops.py:147: add_loss (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.add_loss instead.
---------------------------------------
('Run ID: ', 'cosine-softmax')
('Log directory: ', './output/veri/cosine-softmax')
---------------------------------------
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/contrib/slim/python/slim/learning.py:737: __init__ (from tensorflow.python.training.supervisor) is deprecated and will be removed in a future version.
Instructions for updating:
Please switch to tf.train.MonitoredTrainingSession
2019-02-21 02:46:09.671040: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:864] ARM64 does not support NUMA - returning NUMA node zero
2019-02-21 02:46:09.671174: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1392] Found device 0 with properties: 
name: NVIDIA Tegra X2 major: 6 minor: 2 memoryClockRate(GHz): 1.3005
pciBusID: 0000:00:00.0
totalMemory: 7.67GiB freeMemory: 4.53GiB
2019-02-21 02:46:09.671227: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1471] Adding visible gpu devices: 0
2019-02-21 02:46:10.997518: I tensorflow/core/common_runtime/gpu/gpu_device.cc:952] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-02-21 02:46:10.997589: I tensorflow/core/common_runtime/gpu/gpu_device.cc:958]      0 
2019-02-21 02:46:10.997617: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] 0:   N 
2019-02-21 02:46:10.997788: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1084] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3986 MB memory) -> physical GPU (device: 0, name: NVIDIA Tegra X2, pci bus id: 0000:00:00.0, compute capability: 6.2)
2019-02-21 02:46:29.238167: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at tensor_array_ops.cc:415 : Invalid argument: TensorArray map/TensorArray_1_0: Could not write to TensorArray index 1 because the value shape is [373,413,3] which is incompatible with the TensorArray's inferred element shape: [207,229,3] (consider setting infer_shape=False).
('EnqueueError:', InvalidArgumentError())
2019-02-21 02:46:29.403975: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at tensor_array_ops.cc:415 : Invalid argument: TensorArray map/TensorArray_1_2: Could not write to TensorArray index 1 because the value shape is [608,653,3] which is incompatible with the TensorArray's inferred element shape: [420,277,3] (consider setting infer_shape=False).
('EnqueueError:', InvalidArgumentError())
2019-02-21 02:46:29.560724: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at tensor_array_ops.cc:415 : Invalid argument: TensorArray map/TensorArray_1_4: Could not write to TensorArray index 1 because the value shape is [210,181,3] which is incompatible with the TensorArray's inferred element shape: [219,511,3] (consider setting infer_shape=False).
('EnqueueError:', InvalidArgumentError())
2019-02-21 02:46:29.810188: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at tensor_array_ops.cc:415 : Invalid argument: TensorArray map/TensorArray_1_7: Could not write to TensorArray index 1 because the value shape is [167,169,3] which is incompatible with the TensorArray's inferred element shape: [328,284,3] (consider setting infer_shape=False).
('EnqueueError:', InvalidArgumentError())

**Comment by zengdiqing1994** [comment]
> ## @nwojke Thanks for your excellent work! I trained Deep-SORT model on Jeston TX2 , and this error occurs
> can you help me figure out why this happens? Thank you very much!
> ##############################################
> Train set size: 34267 images, 519 identites
> ('feature dimensionality: ', 128)
> WARNING:tensorflow:From /home/nvidia/cosine_metric_learning/nets/deep_sort/network_definition.py:84: calling l2_normalize (from tensorflow.python.ops.nn_impl) with dim is deprecated and will be removed in a future version.
> Instructions for updating:
> dim is deprecated, use axis instead
> WARNING:tensorflow:From /home/nvidia/cosine_metric_learning/train_app.py:615: sparse_softmax_cross_entropy (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
> Instructions for updating:
> Use tf.losses.sparse_softmax_cross_entropy instead. Note that the order of the logits and labels arguments has been changed.
> WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/contrib/losses/python/losses/loss_ops.py:434: compute_weighted_loss (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
> Instructions for updating:
> Use tf.losses.compute_weighted_loss instead.
> WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/contrib/losses/python/losses/loss_ops.py:147: add_loss (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
> Instructions for updating:
> Use tf.losses.add_loss instead.
> ## ('Run ID: ', 'cosine-softmax')
> ('Log directory: ', './output/veri/cosine-softmax')
> WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/contrib/slim/python/slim/learning.py:737: **init** (from tensorflow.python.training.supervisor) is deprecated and will be removed in a future version.
> Instructions for updating:
> Please switch to tf.train.MonitoredTrainingSession
> 2019-02-21 02:46:09.671040: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:864] ARM64 does not support NUMA - returning NUMA node zero
> 2019-02-21 02:46:09.671174: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1392] Found device 0 with properties:
> name: NVIDIA Tegra X2 major: 6 minor: 2 memoryClockRate(GHz): 1.3005
> pciBusID: 0000:00:00.0
> totalMemory: 7.67GiB freeMemory: 4.53GiB
> 2019-02-21 02:46:09.671227: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1471] Adding visible gpu devices: 0
> 2019-02-21 02:46:10.997518: I tensorflow/core/common_runtime/gpu/gpu_device.cc:952] Device interconnect StreamExecutor with strength 1 edge matrix:
> 2019-02-21 02:46:10.997589: I tensorflow/core/common_runtime/gpu/gpu_device.cc:958] 0
> 2019-02-21 02:46:10.997617: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] 0: N
> 2019-02-21 02:46:10.997788: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1084] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3986 MB memory) -> physical GPU (device: 0, name: NVIDIA Tegra X2, pci bus id: 0000:00:00.0, compute capability: 6.2)
> 2019-02-21 02:46:29.238167: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at tensor_array_ops.cc:415 : Invalid argument: TensorArray map/TensorArray_1_0: Could not write to TensorArray index 1 because the value shape is [373,413,3] which is incompatible with the TensorArray's inferred element shape: [207,229,3] (consider setting infer_shape=False).
> ('EnqueueError:', InvalidArgumentError())
> 2019-02-21 02:46:29.403975: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at tensor_array_ops.cc:415 : Invalid argument: TensorArray map/TensorArray_1_2: Could not write to TensorArray index 1 because the value shape is [608,653,3] which is incompatible with the TensorArray's inferred element shape: [420,277,3] (consider setting infer_shape=False).
> ('EnqueueError:', InvalidArgumentError())
> 2019-02-21 02:46:29.560724: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at tensor_array_ops.cc:415 : Invalid argument: TensorArray map/TensorArray_1_4: Could not write to TensorArray index 1 because the value shape is [210,181,3] which is incompatible with the TensorArray's inferred element shape: [219,511,3] (consider setting infer_shape=False).
> ('EnqueueError:', InvalidArgumentError())
> 2019-02-21 02:46:29.810188: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at tensor_array_ops.cc:415 : Invalid argument: TensorArray map/TensorArray_1_7: Could not write to TensorArray index 1 because the value shape is [167,169,3] which is incompatible with the TensorArray's inferred element shape: [328,284,3] (consider setting infer_shape=False).
> ('EnqueueError:', InvalidArgumentError())

老哥，你这问题解决了吗？能否加个联系方式？

**Comment by nwojke** [comment]
The important line seems to be the following, indicating a shape mismatch for the input images. I have trained on VeRi yesterday without issues. Which TensorFlow version are you using?

```
2019-02-21 02:46:29.810188: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at tensor_array_ops.cc:415 : Invalid argument: TensorArray map/TensorArray_1_7: Could not write to TensorArray index 1 because the value shape is [167,169,3] which is incompatible with the TensorArray's inferred element shape: [328,284,3] (consider setting infer_shape=False).
```

**Comment by TongJiL** [comment]
Could you please also provide the trained results for VeRi? It would be a really big help to me and a lot of people.

**Comment by nwojke** [comment]
We are currently not making the VeRi model public because the dataset requires signing an agreement form. The code that we provide can be used to train on VeRi without modifications if you have the data available. Therefore, you should be able to reproduce our (non-published) results as soon as you have access to the data.

In this light, I currently don't see the use-case for releasing trained models.

**Comment by LARAWXL** [comment]
@nwojke Hi, I face this problem as well InvalidArgumentError (see above for traceback): TensorArray map/TensorArray_1_6: Could not write to TensorArray index 1 because the value shape is [86,76,3] which is incompatible with the TensorArray's inferred element shape: [125,143,3] (consider setting infer_shape=False).

I think this may be because the VeRi dataset has a different size of images to train. But Market1501 and Mars have exactly the same resolution (64*128) images so the TensorArray that they generated have the same shape and will not have this problem. Although I used cv2.resize() to force the VeRi images to resize to 64*128. It still raises this issue.

I am using tensorflow1.10 version + cuda9.0

You said you have successfully applied the code on VeRi. Could you please provide the detail of it? For example, which .py files were you use. train_market1501.py, train_mars.py, market1501.py or mars.py? Also, I wonder whether you have changed any code about resizing the images or other things. 

If possible, could you kindly share your any piece of edited code (like train_VeRi.py and dataset/VeRi.py) for VeRi dataset with me? Thank you so much!

**Comment by nwojke** [comment]
Are you sure that you didn't make any changes to the code in branch ``feature/veri_dataset``? I haven't made any changes and trained on VeRi when I made my comment on April 30.

You are right that VeRi has images of different sizes, but the code reshapes all images into width 128 and height 64. This happens [here](https://github.com/nwojke/cosine_metric_learning/blob/feature/veri_dataset/datasets/veri.py#L67). The function is called from ``train_veri.py`` [here](https://github.com/nwojke/cosine_metric_learning/blob/feature/veri_dataset/train_veri.py#L55). Did you maybe change this code to call ``read_train_split_to_str()``? This could explain the error.

**Comment by JIALIUZHOU** [comment]
Hi, I would like to ask the question whether you have solved the problem? I met the same problem as yours when I use the mode 'export''. And I do not know how to solve it.

**Comment by LARAWXL** [comment]
> Are you sure that you didn't make any changes to the code in branch `feature/veri_dataset`? I haven't made any changes and trained on VeRi when I made my comment on April 30.
> 
> You are right that VeRi has images of different sizes, but the code reshapes all images into width 128 and height 64. This happens [here](https://github.com/nwojke/cosine_metric_learning/blob/feature/veri_dataset/datasets/veri.py#L67). The function is called from `train_veri.py` [here](https://github.com/nwojke/cosine_metric_learning/blob/feature/veri_dataset/train_veri.py#L55). Did you maybe change this code to call `read_train_split_to_str()`? This could explain the error.

Thank you very much. Didnt see the branch XD. Problem solved. Dont know the reason tho.

**Comment by LARAWXL** [comment]
> Hi, I would like to ask the question whether you have solved the problem? I met the same problem as yours when I use the mode 'export''. And I do not know how to solve it.

I didnt encounter since I used the VeRi branch. Have you used it?

**Comment by varunvelugoti** [comment]
> The important line seems to be the following, indicating a shape mismatch for the input images. I have trained on VeRi yesterday without issues. Which TensorFlow version are you using?
> 
> ```
> 2019-02-21 02:46:29.810188: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at tensor_array_ops.cc:415 : Invalid argument: TensorArray map/TensorArray_1_7: Could not write to TensorArray index 1 because the value shape is [167,169,3] which is incompatible with the TensorArray's inferred element shape: [328,284,3] (consider setting infer_shape=False).
> ```

Hi nwojke
Could you tell me which version of tensorflow you're using when trained on Veri Dataset ?
